### PR TITLE
Clarified database connections lifetime outside HTTP requests.

### PR DESCRIPTION
**Pull request by albertosottile**
At first glance, I thought this section (and the whole page) referred to requests to the database, hinting that Django would have recreated connections to the database after each query.

In reality, Django manages that only in the context of `HttpRequest` calls. Hence, this commit clarifies this behavior in the documentation.

/cc @schinckel

### CODE REVIEW

**Comment by schinckel** [comment]
Someone else may want to chime in, but I’m not sure if we need the link: just stating that request means HTTP request-response cycle may be enough. 

Either that, or being even more explicit. It’s too late here now for me to think further about it.

**Comment by albertosottile** [comment]
I have no strong feelings one way or the other. I wanted to leave it more generic but, I was afraid it would not have clarified the concept enough. 

Please let me know what formulation you would prefer.

**Comment by nessita** [comment]
Hello! I'm doing some PR cleanup. Do you have time to keep working on this? If yes, would you consider updating it with the latest `main` branch, resolving conflicts, and flag this PR for review? Thank you!

**Comment by albertosottile** [comment]
I will be away from home until the end of next week. Can I update the pull request after?

**Comment by nessita** [comment]
> I will be away from home until the end of next week. Can I update the pull request after?

Yes, absolutely! Thanks.

**Comment by albertosottile** [comment]
@nessita I rebased the PR, but I do not know how to "flag it for review". Could you take care of it from here? Thank you

File: docs/ref/databases.txt [hunk 1]
```diff
@@ -97,6 +97,11 @@ either restore Django's defaults at the end of each request, force an
 appropriate value at the beginning of each request, or disable persistent
 connections.
 
+If you use database connections outside of the context of
+:class:`~django.http.HttpRequest` requests, Django will not close the
+connection and will keep using the connection created after the first database
+query.
```
Comment (sarahboyce) [inline docs/ref/databases.txt:103]: I found this addition a little confusing because it made me think.... does this mean the connection **never** gets closed?

So outside a request response cycle, we are probably talking about things like management commands (the shell included). At the end of a management command, django closes all of the database connections - you can see this here: https://github.com/django/django/blob/2b11740e1d9d23ef53d94a1c45896fb7b91df908/django/core/management/base.py#L425
I'm not sure how to rephrase this to make it clear that people shouldn't panic that they need to manually close database connections when they have finished doing something in the shell for example.

What do you think?

Comment (albertosottile) [reply docs/ref/databases.txt:103]: > So outside a request response cycle, we are probably talking about things like management commands (the shell included).

That depends, what sparked this PR was me using the Django ORM in some Python logic, without ever going through a web server/WSGI.

> At the end of a management command, django closes all of the database connections - you can see this here:

Does this apply also to manual uses of the ORM (e.g. queries made with a `Manager`, e.g. `objects`)? Because this does not match my experience (the connection was created only at the beginning of the Python logic and never closed/reopened until the end of the process).

> What do you think?

If your concern is just the management shell, I think we can add an extra sentence to explicitly rule that out. But I think we should first make sure we agree on the actual issue addressed by this PR.

Comment (sarahboyce) [reply docs/ref/databases.txt:103]: > queries made with a Manager, e.g. objects

No sorry I didn't mean the model manager. I meant when you run `manage.py <some-example-management-command>` (eg. shell or migrate - there are quite a lot that django provides and [you can also write your own](https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/)). 

So your experience is correct :+1:

> and never closed/reopened until the end of the process

:point_up: This is what I am trying to refer to, at the end, django closes the connection

> me using the Django ORM in some Python logic

Can you be more specific? Where/how does that Python logic get run/triggered?

Comment (sarahboyce) [reply docs/ref/databases.txt:103]: Ah so in your case it not a django project, then what I said about it closing at the end around management commands does not apply and yes django wouldn't help you - I'm on the same page (sorry took me some time to get there)

---
In my opinion, as this is not the standard use case, this paragraph can cause confusion. If you reduce the docs to just the http request references, I am happy to approve. :+1: 

If you want the additional paragraph, I would like to see how common this use case is. If it's common enough, maybe we would even need a topic like "Using the django orm as a standalone" and the setup and comments around connections can be described in there. We could even add features to make this easier :woman_shrugging: 

Personally, I want to see either community input that this fills some documentation gap (that doesn't belong in a blog post) and an accepted ticket to add this into the docs. :heavy_check_mark: 

But that's just my opinion, so I would be interested in what @nessita and @felixxm think?

Comment (albertosottile) [reply docs/ref/databases.txt:103]: I am fine with that. IMHO the point of the PR was just to somehow provide a warning that Django does not really manage the DB connection for you outside of the standard use case. If you can find a better formulation for this statement, I am more than willing to update the PR.

Comment (nessita) [reply docs/ref/databases.txt:103]: Thank you @sarahboyce and @albertosottile for the many clarifications around the changes.

Considering the above, I would suggest that we:

1- Remove the link to HttpRequest as originally request and perhaps say "HTTP request" the first time "request" is mentioned.

2- We add a small and concise paragraph similar to the one in https://docs.djangoproject.com/en/4.2/topics/db/transactions/#controlling-transactions-explicitly:

```suggestion
If a connection is created in a long-running process, outside of Django’s
request / response cycle, the connection will remain open until explicitly
closed.
```

File: docs/ref/databases.txt [hunk 2]
```diff
@@ -31,14 +31,14 @@ Persistent connections
...
+controlled by the :setting:`CONN_MAX_AGE` parameter which defines the maximum
+lifetime of a connection. It can be set independently for each database.
 
 The default value is ``0``, preserving the historical behavior of closing the
-database connection at the end of each request. To enable persistent
-connections, set :setting:`CONN_MAX_AGE` to a positive integer of seconds. For
-unlimited persistent connections, set it to ``None``.
+database connection at the end of each :class:`~django.http.HttpRequest`. To
+enable persistent connections, set :setting:`CONN_MAX_AGE` to a positive
+integer of seconds. For unlimited persistent connections, set it to ``None``.
```
Comment (nessita) [inline docs/ref/databases.txt:41]: I would undo all these changes and just have the first reference to `request` be `HTTP request`.

File: docs/ref/databases.txt [hunk 3]
```diff
@@ -97,6 +97,10 @@ either restore Django's defaults at the end of each request, force an
 appropriate value at the beginning of each request, or disable persistent
 connections.
 
+If a connection is created in a long-running process, outside of Django’s
+request / response cycle, the connection will remain open until explicitly
```
Comment (felixxm) [inline docs/ref/databases.txt:101]:
```suggestion
request-response cycle, the connection will remain open until explicitly
```

**Comment by felixxm** [comment]
@albertosottile Thanks :+1: Welcome aboard :boat:

**Comment by albertosottile** [comment]
Thank you for merging this!
